src/main/java/org/zalando/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -68,19 +75,19 @@ private void onTopologyChanged() {
             throw new IllegalStateException(
                     "topologyListener should not be null when calling onTopologyChanged method");
         }
-        topologyListener.refresh();
         reactOnTopologyChange();
     }
 
-    private void reactOnTopologyChange() {
+    private void reactOnTopologyChange() throws NakadiRuntimeException {
+        final ZkSubscriptionClient.Topology topology = topologyListener.getData();
 
         // Collect current partitions state from Zk
         final Map<EventTypePartition, Partition> partitions = new HashMap<>();
-        getZk().runLocked(() -> Stream.of(getZk().listPartitions())
```

Comment:
Is it the call to ZK which was removed?

Reply:
no, i will add comment where the actual stuff happend